pkg/controller/node/node_controller.go

```diff
@@ -622,15 +622,18 @@ func (ctrl *Controller) getPoolsForNode(node *corev1.Node) ([]*mcfgv1.MachineCon
 	if len(custom) > 1 {
 		return nil, fmt.Errorf("node %s belongs to %d custom roles, cannot proceed with this Node", node.Name, len(custom))
 	} else if len(custom) == 1 {
-		// We don't support making custom pools for masters
+		pls := []*mcfgv1.MachineConfigPool{}
 		if master != nil {
-			return nil, fmt.Errorf("node %s has both master role and custom role %s", node.Name, custom[0].Name)
+			// if we have a custom pool and master, defer to master and return.
+			glog.Infof("Given both master and custom pools. Defaulting to master: custom %v", custom[0].Name)
```

Comment:
I think this makes sense, but I think we should probably have an alert or error somehow to inform the user that the custom pool they are trying to apply is for a master node, which is a no-op.

Although we fail today, I can see how users would be tripped up if they tried to do this, and wondering why the custom pool configs are not supplied, if all we have is a info level logging

Reply:
do you mean a Prometheus alert?

Reply:
Yes, I think so, a `warning` level alert without degrading the pool is probably the best way to get the user's attention